src/xwayland/xwm.c

```diff
@@ -209,7 +209,11 @@ read_properties(struct wlc_xwm *xwm, struct wlc_x11_window *win, const xcb_atom_
             size_t class_total_len = xcb_get_property_value_length(reply);
 
             size_t class_class_offset = strnlen(class, class_total_len)+1;
-            wlc_view_set_class_ptr(view, class+class_class_offset, class_total_len-class_class_offset);
+            if (class_class_offset >= class_total_len) {
+               wlc_view_set_class_ptr(view, class, class_class_offset-1);
```

Comment:
This'd make for either an additional variable or an additional `strlen()` call, it doesn't really matter though